src/snowflake/snowpark/session.py

```diff
@@ -247,6 +249,19 @@ class SessionBuilder:
 
         def __init__(self) -> None:
             self._options = {}
+            self.__appname__ = None
+
+        def appName(self, name: str) -> "Session.SessionBuilder":
```

Comment:
The implementation uses query tag. A user may replace the query tag so it may not be reliable.